src/gfx/process.cpp

```diff
@@ -953,8 +965,56 @@ static UniqueTiles dedupTiles(
 	// by caching the full tile data anyway, so we might as well.)
 	UniqueTiles tiles;
 
+	if (!options.inputTileset.empty()) {
+		File inputTileset;
+		if (!inputTileset.open(options.inputTileset, std::ios::in | std::ios::binary)) {
+			fatal("Failed to open \"%s\": %s", options.inputTileset.c_str(), strerror(errno));
+		}
+
+		std::array<uint8_t, 16> tile;
+		size_t const tileSize = options.bitDepth * 8;
+		for (;;) {
+			// It's okay to cast between character types.
+			size_t len = inputTileset->sgetn(reinterpret_cast<char *>(tile.data()), tileSize);
+			if (len == 0) { // EOF!
+				break;
+			} else if (len != tileSize) {
+				fatal(
+				    "\"%s\" does not contain a multiple of %zu bytes; is it actually tile data?",
+				    options.inputTileset.c_str(),
+				    tileSize
+				);
+			} else if (len == 8) {
+				// Expand the tile data to 2bpp.
```

Comment:
...Or maybe it doesn't? If not, why not? I'd expect that if I pass `-d 1 -i foo.1bpp`, that should work, and if I pass `-d 2 -i foo.2bpp` (or just `-i foo.2bpp` since 2 is default) then it should just work. Of course, that would allow for accidentally interpreting a 1bpp input tileset as a 2bpp one of half the size, but that's GIGO again. ("Pray, Mr. Babbage, if you put into the machine wrong figures, will the right answers come out?")

Reply:
Both work. This code path is only reached if we have read one full tile (the first `if` handles EOF, the second handles any incomplete read) and the bit depth is 1bpp.

It translates the 1bpp data into 2bpp, since the rest of RGBGFX processes 2bpp data (depth is otherwise only applied during palette packing (as a limit to palette size) and tile data emission (skipping every other bitplane).

Reply:
I think this makes sense. Then some of the above docs should avoid mentioning 2bpp specifically+only.